QSPIFBlockDevice.cpp

```diff
@@ -0,0 +1,1263 @@
+SingletonPtr<PlatformMutex> QSPIFBlockDevice::_mutex;
+
+// Local Function
+int localMathPower(int base, int exp);
+
+/********* Public API Functions *********/
+/****************************************/
+
+QSPIFBlockDevice::QSPIFBlockDevice(PinName io0, PinName io1, PinName io2, PinName io3, PinName sclk, PinName csel,
+                                   qspif_polarity_mode clock_mode, int freq)
+    : _qspi(io0, io1, io2, io3, sclk, csel, clock_mode), _device_size_bytes(0)
+{
+    _is_initialized = false;
+    _min_common_erase_size = 0;
+    _regions_count = 1;
+    _region_erase_types_bitfield[0] = ERASE_BITMASK_NONE;
+
+    //Default Bus Setup 1_1_1 with 0 dummy and mode cycles
+    _inst_width = QSPI_CFG_BUS_SINGLE;
+    _address_width = QSPI_CFG_BUS_SINGLE;
+    _address_size = QSPI_CFG_ADDR_SIZE_24;
+    _data_width = QSPI_CFG_BUS_SINGLE;
+    _dummy_and_mode_cycles = 0;
+
+    if (QSPI_STATUS_OK != _qsp_set_frequency(freq)) {
+        QSPIF_LOG(QSPIF_DEBUG_ERROR, "ERROR: QSPI Set Frequency Failed");
+    }
+}
+
+
+int QSPIFBlockDevice::init()
+{
+
+    uint8_t vendor_device_ids[4];
+    size_t data_length = 3;
+    int status = QSPIF_BD_ERROR_OK;
+    uint32_t basic_table_addr = NULL;
+    size_t basic_table_size = 0;
+    uint32_t sector_map_table_addr = NULL;
+    size_t sector_map_table_size = 0;
+    int qspi_status = QSPI_STATUS_OK;
+
+    _mutex->lock();
+    if (_is_initialized == true) {
+        goto Exit_Point;
+    }
+
+    // Soft Reset
+    if ( -1 == _reset_flash_mem()) {
+        QSPIF_LOG(QSPIF_DEBUG_INFO, "ERROR: init - Unable to initialize flash memory, tests failed\n");
+        status = QSPIF_BD_ERROR_DEVICE_ERROR;
+        goto Exit_Point;
+    } else {
+        QSPIF_LOG(QSPIF_DEBUG_INFO, "INFO: Initialize flash memory OK\n");
+    }
+
+    /* Read Manufacturer ID (1byte), and Device ID (2bytes)*/
+    qspi_status = _qspi_send_read_command(QSPIF_RDID, (char *)vendor_device_ids, 0x0 /*address*/, data_length);
+    if (qspi_status != QSPI_STATUS_OK) {
+        QSPIF_LOG(QSPIF_DEBUG_ERROR, "ERROR: init - Read Vendor ID Failed");
+        status = QSPIF_BD_ERROR_DEVICE_ERROR;
+        goto Exit_Point;
+    }
+
+    switch (vendor_device_ids[0]) {
+        case 0xbf:
+            // SST devices come preset with block protection
+            // enabled for some regions, issue write disable instruction to clear
+            _set_write_enable();
+            _qspi_send_general_command(QSPIF_WRDI, -1, NULL, 0, NULL, 0);
+            break;
+    }
+
+    //Synchronize Device
+    if ( false == _is_mem_ready()) {
+        QSPIF_LOG(QSPIF_DEBUG_ERROR, "ERROR: init - _is_mem_ready Failed");
+        status = QSPIF_BD_ERROR_READY_FAILED;
+        goto Exit_Point;
+    }
+
+    /**************************** Parse SFDP Header ***********************************/
+    if ( 0 != _sfdp_parse_sfdp_headers(basic_table_addr, basic_table_size, sector_map_table_addr, sector_map_table_size)) {
```

Comment:
`if (0...` space should not be there around `(text)`

Reply:
fixed